strapi/strapi
# Typescript/toolchain

## What does it do?

### Toolchain

- Handle typescript for the `build` & `develop` commands
  - `build` compile all the files into a dist folder
  - `develop` compile all the files into a dist folder & watch for any change in the compiled files (then rebuild, etc...)

### Core
- Allow loading default exports

### CLI
- Add `ts` and `typescript` flags to the project generator. Currently, it only adds a new attribute in the create-project scopes, but it'll eventually be used to create a project with TS files in the next PR. 

## Why is it needed?

Needed to initialize the support of projects written in typescript.

## How to test it?
 - `yarn setup`
 - `cd examples/kitchensink-ts`
 - `yarn build`
 - `yarn start dist`
 - `yarn develop` (then change any ts file to see the server recompile & restart)

## Known Issues
- The develop command cannot be launched without a first TS build. This is due to a logger instance being created by loading the config before actually starting the TS compiler
